plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -59,10 +69,12 @@ func (*OpenTelemetry) SampleConfig() string {
 
 func (o *OpenTelemetry) Connect() error {
 	logger := &otelLogger{o.Log}
-
 	if o.ServiceAddress == "" {
 		o.ServiceAddress = defaultServiceAddress
 	}
+	if o.EncodingType == "" {
+		o.EncodingType = "application/x-protobuf"
+	}
```

Comment:
Can we please check the encoding type!? What if some uses `application/pdf`? I guess we are only support JSON and protobuf for now, don't we?

Use a switch statement like

```golang
	switch o.EncodingType {
	case "":
		o.EncodingType = "application/x-protobuf"
	case "application/json", "application/x-protobuf":
		// Do nothing, those are valid
	default:
		return fmt.Errorf("invalid encoding %q", o.EncodingType)
	}
```

to be honest I would even prefer

```golang
	switch o.EncodingType {
	case "", "protobuf":
		o.EncodingType = "application/x-protobuf"
	case "json":
		o.EncodingType = "application/json"
	default:
		return fmt.Errorf("invalid encoding %q", o.EncodingType)
	}
```

Reply:
Applied.
https://github.com/influxdata/telegraf/pull/17997/commits/c981b1591845b89ee7e37edcc4c1e13ceae77ce8